File: android/src/main/kotlin/com/hisaichi5518/native_webview/WebViewManager.kt
package com.hisaichi5518.native_webview

import androidx.webkit.WebViewCompat
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler

class WebViewManager(messenger: BinaryMessenger) : MethodCallHandler {

    private var channel = MethodChannel(messenger, "com.hisaichi5518/native_webview_webview_manager")

    init {
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: MethodChannel.Result) {
        when (call.method) {
            "getAndroidWebViewInfo" -> {
                val context = Locator.activity?.applicationContext
                if (context == null) {
                    result.success(null)
                    return
                }
                val packageInfo = WebViewCompat.getCurrentWebViewPackage(context)
                if (packageInfo == null) {
                    result.success(null)
                    return
                }

                result.success(mapOf(
                    "versionName" to packageInfo.versionName,
                    "packageName" to packageInfo.packageName
                ))
            }
            else -> result.notImplemented()
        }
    }

    fun dispose() {
        channel.setMethodCallHandler(null)
    }
}

File: ios/Classes/WebViewManager.swift
//
//  WebViewManager.swift
//  native_webview
//
//  Created by hisaichi5518 on 2020/05/21.
//

import Foundation
import WebKit
import Flutter

class WebViewManager: NSObject, FlutterPlugin {
    var channel: FlutterMethodChannel?

    static func register(with registrar: FlutterPluginRegistrar) {
        let manager = WebViewManager()
        manager.channel = FlutterMethodChannel(name: "com.hisaichi5518/native_webview_webview_manager", binaryMessenger: registrar.messenger())
        registrar.addMethodCallDelegate(manager, channel: manager.channel!)
    }

    public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
        switch call.method {
            default:
                result(FlutterMethodNotImplemented)
        }
    }
}
